extensions/wikia/Recirculation/RecirculationHooks.class.php

```diff
@@ -98,19 +97,54 @@ static public function canShowDiscussions( $cityId ) {
 		}
 	}
 
-	private static function addMainPageMetadata( OutputPage $outputPage ) {
-		if ( F::app()->wg->Title->isMainPage() ) {
-			$promoDetails = WikiaDataAccess::cache(
-					wfMemcKey( "site-attribute-liftigniterMetadata" ),
-					3600, // one hour cache
-					function() {
-						global $wgCityId;
-						return ( new SiteAttributeService() )->getAttribute( $wgCityId, "liftigniterMetadata" );
-					} );
-
-			if ( $promoDetails !== null ) {
-				$outputPage->addScript( "<script id=\"liftigniter-metadata\" type=\"application/json\">${promoDetails}</script>" );
+	private static function getLiftIgniterMetadataFromSiteAttributeService() {
+		return WikiaDataAccess::cache(
+			wfMemcKey( 'site-attribute-liftigniterMetadata' ),
+			3600,
+			function () {
+				global $wgCityId;
+
+				$decodedData = [];
+				$rawData = ( new SiteAttributeService() )
+					->getAttribute( $wgCityId, 'liftigniterMetadata' );
+
+				if ( !empty( $rawData ) ) {
+					try {
+						$decodedData = json_decode( $rawData, true );
+					} catch ( Exception $exception ) {}
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS108)